mkosi/__init__.py

```diff
@@ -4368,24 +4368,37 @@ def generate_key_cert_pair(args: Args) -> None:
 
 
 def bump_image_version() -> None:
-    """Write current image version plus one to mkosi.version"""
-    version = Path("mkosi.version").read_text().strip()
-    v = version.split(".")
+    """
+    Read current version from 'mkosi.version' and run 'mkosi.bump'
+    or increase version plus one and write it back to 'mkosi.version'
+    """
 
-    try:
-        m = int(v[-1])
-    except ValueError:
-        new_version = version + ".2"
-        logging.info(
-            "Last component of current version is not a decimal integer, "
-            f"appending '.2', bumping '{version}' → '{new_version}'."
-        )
+    if not (version_file := Path("mkosi.version")).exists():
+        die(f"Cannot bump image version, '{version_file}' not found")
+
+    version = version_file.read_text().strip()
+
+    if (script := Path("mkosi.bump").absolute()).exists():
+        if not os.access(script, os.X_OK):
+            die(f"{script} is not executable")
+
+        with complete_step(f"Running bump script {script}…"):
+            new_version = run([script, version], stdout=subprocess.PIPE).stdout.strip()
```

Comment:
This should have some sandboxing so that it cannot write anything, with all the script running moved to a separate `run_bump_script` function.